sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/PushDownUtils.scala

```diff
@@ -50,8 +49,17 @@ object PushDownUtils extends PredicateHelper {
         val translatedFilters = mutable.ArrayBuffer.empty[sources.Filter]
         // Catalyst filter expression that can't be translated to data source filters.
         val untranslatableExprs = mutable.ArrayBuffer.empty[Expression]
+        val dataFilters = r match {
+          case f: FileScanBuilder =>
+            val (partitionFilters, fileDataFilters) =
+              DataSourceUtils.getPartitionKeyFiltersAndDataFilters(
+              f.getSparkSession, scanBuilderHolder.relation, f.readPartitionSchema(), filters)
+            f.pushPartitionFilters(ExpressionSet(partitionFilters).toSeq, fileDataFilters)
```

Comment:
can we make it clear what's the contract?

I thought we add this private API `pushPartitionFilters` for file source v2, which is only used to push partition filters. Individual file source v2 implementation can implement `SupportsPushDownFilters` to get data filters.

Reply:
As per our offline discussion, I have made the following changes:
- make file source v2 NOT implement `SupportsPushdownFilters` any more
- add `pushFiltersToFileIndex` in file source v2. In this method:
    - push both Expression partition filters and Expression data filters to file source.
    - data filters are used for filter push down. File source translates the data filters from `Expression` to `Sources.Filer`, and decides which filters to push down.
   - partition filters are used for partition pruning.

I have updated the PR description accordingly.